### PR TITLE
Close Nav Menu on Click

### PR DESCRIPTION
**Pull request by dbalders**
Partial of #5483
- Creates new gh-nav-list component
- Component called when the nav li's are clicked
- Uses the closeMenus function to close the menu when clicked

### CODE REVIEW

File: core/client/app/templates/components/gh-nav-menu.hbs [hunk 1]
```diff
@@ -20,22 +20,26 @@
...
     {{#if (gh-user-can session.user 'admin')}}
         <ul class="gh-nav-list gh-nav-settings">
             <li class="gh-nav-list-h">Settings</li>
-            <li>{{#link-to "settings.general" classNames="gh-nav-settings-general"}}<i class="icon-settings"></i>General{{/link-to}}</li>
-            {{!<li><i class="icon-design"></i>Themes</li>}}
-            <li>{{#link-to "settings.navigation" classNames="gh-nav-settings-navigation"}}<i class="icon-compass"></i>Navigation{{/link-to}}</li>
-            <li>{{#link-to "settings.tags" classNames="gh-nav-settings-tags"}}<i class="icon-tag"></i>Tags{{/link-to}}</li>
-            <li>{{#link-to "settings.code-injection" classNames="gh-nav-settings-code-injection"}}<i class="icon-code"></i>Code Injection{{/link-to}}</li>
-            <li>{{#link-to "settings.labs" classNames="gh-nav-settings-labs"}}<i class="icon-apps"></i>Labs{{/link-to}}</li>
+            {{#gh-nav-list onClick="closeMenus"}}
```
Comment (novaugust) [inline core/client/app/templates/components/gh-nav-menu.hbs]: You're kind of breaking html rules by putting a div inside of a ul. Probably better off just having this thing take the place of the ul itself, right?

Comment (dbalders) [inline core/client/app/templates/components/gh-nav-menu.hbs]: Should I try and do that, or maybe just wrap the whole thing in the `{{#gh-nav-list onClick="closeMenus"}}` and try and target the li's on the click? Whatever you think is better I can do.

Comment (jaswilli) [inline core/client/app/templates/components/gh-nav-menu.hbs]: You can likely just ditch the component altogether assign the action the ul element itself:

`<ul {{action "closeMenus}} class="gh-nav-list gh-nav-settings">`

Comment (dbalders) [inline core/client/app/templates/components/gh-nav-menu.hbs]: that works! I would rather it not happen on the headers though, should I just add that action to each li itself?

Comment (jaswilli) [inline core/client/app/templates/components/gh-nav-menu.hbs]: I wonder if we can just move the header out into it's own list:

```
<ul class="gh-nav-list">
    <li class="gh-nav-list-h">Settings</li>
</ul>
<ul {{action "closeMenus"}} class="gh-nav-list gh-nav-settings">
    <li>{{#link-to}}...{{/link-to}}</li>
</ul>

**Comment by dbalders** [comment]
@jaswilli: ready using your method if you wanted to take another look.

**Comment by dbalders** [comment]
Fixed in #5699

**Pull request closed without merging**